src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -370,6 +442,10 @@ CHIP_ERROR ConfigureState(chip::FabricIndex fabricIndex, chip::NodeId nodeId)
             ResetState();
         }
 
+        // Start a timer to track whether we receive a BDX init after a successful query image in a reasonable amount of time
+        CHIP_ERROR err = mSystemLayer->StartTimer(kBdxInitReceivedTimeout, HandleBdxInitReceivedTimeoutExpired, this);
+        VerifyOrReturnError(err == CHIP_NO_ERROR, CHIP_ERROR_INCORRECT_STATE);
```

Comment:
Why do we lose the original error here? why return CHIP_ERROR_INCORRECT_STATE regardless of real error?

Why not `ReturnErrorOnFailure(err)` ?

Reply:
The caller maps CHIP_ERROR_INCORRECT_STATE to specific bdx responses, and we don't want to try to guess at all the possible errors StartTimer can return.

Reply:
That said, that's worth documenting here so this code does not get changed.

Reply:
added a comment here as well

Reply:
In that case we should log the original error for debugging at least, since on larger platforms that gives us the real error and maybe even line information. Having `INCORRECT_STATE at DelegateBridge:450` is not that useful as the real StartTimer Error.

Reply:
Also were is the caller code mapping this to BDX errors? Can't it have the INCORRECT_STATE as a default or is there some other default?

I guess documentation then has to say that this method MUST always return a subset of errors. Is that documented?

Reply:
and going further ... if the caller needs a BDX state not a chip error (i.e. it has to map) ... why does this function return a CHIP_ERROR at all?

Reply:
There's a lot of general logic cleanup that should happen here, but we are trying to avoid scope creep... Will do that in a followup.

Reply:
i added the logerroronfailure to address - "
In that case we should log the original error for debugging at least, since on larger platforms that gives us the real error and maybe even line information. Having INCORRECT_STATE at DelegateBridge:450 is not that useful as the real StartTimer Error." The rest I am going to address as a follow up in the interest of time. @andy31415 hope thats ok

Reply:
Filed - https://github.com/project-chip/connectedhomeip/issues/24859 to track -  if the caller needs a BDX state not a chip error (i.e. it has to map) ... why does this function return a CHIP_ERROR at all? and There's a lot of general logic cleanup that should happen here, but we are trying to avoid scope creep... Will do that in a followup.